src/node.rs

```diff
@@ -265,13 +315,16 @@ impl PbftNode {
         }
     }
 
-    fn check_if_checkpoint_started(&mut self, msg: &PeerMessage) -> bool {
+    fn check_if_checkpoint_started(&mut self, msg: &PeerMessage, sender_id: &PeerId) -> bool {
```

Comment:
Haven't we already rejected the message by the time `check_if_checkpoint_started` is called? Wondering if we can avoid pushing messages to the backlog that have invalid sender IDs, since I assume them having those invalid IDs won't change in the future

Reply:
That's correct, the message should already be rejected before this point; no messages with invalid sender IDs will get pushed to the backlog. However, we still need to hold onto that sender ID since `on_peer_message` requires it. If we don't want to have to keep the sender ID at this point, we would have to refactor the main functionality of `on_peer_message` behind another function that doesn't need the sender ID and essentially just check the sender ID in `on_peer_message` itself.

Reply:
Ah cool, makes sense. I think as a companion to https://bitwiseio.atlassian.net/browse/IP-72, we could refactor the protobuf decoding to happen in the `engine.rs` loop, and reject messages with bad sender IDs before they even hit `on_peer_message`. That's also quite out-of-scope for this PR though, so ✅